src/lib/components/Notifier.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+    import { onMount, tick } from "svelte";
+
+    let showNotice = false
+    let permissionGranted = false
+
+    async function requestPermission() {
+        showNotice = true
+        await tick() // show the notice before we requrest the permission
+        const permission = await Notification.requestPermission()
+        showNotice = false
+        return permission
+    }
+    
+    onMount(async () => {
+        if (!('Notification' in window)) {
+            console.log("This browser does not support the Notification API")
+        } else {
+            try {
+                if (Notification.permission === 'default') {
+                    permissionGranted = await requestPermission() === 'granted' ? true : false
+                } else if (Notification.permission === 'granted') {
+                    await Notification.requestPermission() // let the browser know we'll use it
+                    permissionGranted = true
+                }
+                console.log(permissionGranted)
+            } catch (e) {
+                showNotice = false
+                console.log("This browser does not support the promisified Notification API")
+            }
+        }
+    })
+
+    
+</script>
+
+<script context="module" lang="ts">
+    export function timerEndNotification(timerLength: string) {
+        new Notification('niagara', {
+            body: `Time's up! Your timer for ${timerLength} has gone off.`,
```

Comment:
Can you change `${timerLength}` so the time (5 sec, 00:00:05, 5 seconds) are consistent every time?

Reply:
i don't really know how i would do this. maybe run it through the formatTime function? if so, how?

Reply:
You can use a modified version of `convertTime` to convert the input to seconds, and convert that to 00:00:00 format with `formatTime()`

Reply:
i tried to make sense of the `inputParser` function but it is not commented at all and i don't quite understand it. if you could either explain briefly what does what or write up some code to do what you want here, that would be great.
thanks.

Reply:
`inputParser()` just converts the input string to seconds so you can just do `inputParser(timerLength)` to get the time in seconds. Then, you can use `formatTime()` to convert the seconds to 00:00:00 format, so combining it all:

```ts
const timerLengthSeconds = inputParser(timerLength)
const formattedTimerLength = formatTime(timerLengthSeconds)
```

Reply:
added this